Compatibility with chat UIs (BetterChatGPT)

I succesfully deployed the worker and confirmed it does work with [OpenAI's API reference examples](https://platform.openai.com/docs/api-reference/chat) but running a [fork](https://github.com/chitudorin/chitudorin.github.io) of [BetterChatGPT](https://github.com/ztjhz/BetterChatGPT) (in order to permit newer models) does not work.  

Running log stream on my worker returns a 404 error and it looks like the request is an OPTIONS one, instead of GET.

I tried deploying it but there are some build errors:

```
$ yarn deploy
yarn run v1.22.22
$ wrangler deploy index.js
 ⛅ wrangler 3.37.0
-------------------

✘ [ERROR] No matching export in "node_modules/itty-router/dist/itty-router.min.mjs" for import "createCors"

    index.js:1:17:
      1 │ import { Router, createCors, error, json } from 'itty-router';
        ╵                  ~~~~~~~~~~


✘ [ERROR] No matching export in "node_modules/itty-router/dist/itty-router.min.mjs" for import "error"

    index.js:1:29:
      1 │ import { Router, createCors, error, json } from 'itty-router';
        ╵                              ~~~~~


✘ [ERROR] No matching export in "node_modules/itty-router/dist/itty-router.min.mjs" for import "json"

    index.js:1:36:
      1 │ import { Router, createCors, error, json } from 'itty-router';
        ╵                                     ~~~~


✘ [ERROR] Build failed with 3 errors:

  index.js:1:17: ERROR: No matching export in "node_modules/itty-router/dist/itty-router.min.mjs"
  for import "createCors"
  index.js:1:29: ERROR: No matching export in "node_modules/itty-router/dist/itty-router.min.mjs"
  for import "error"
  index.js:1:36: ERROR: No matching export in "node_modules/itty-router/dist/itty-router.min.mjs"
  for import "json"

```

--- a/.gitignore
+++ b/.gitignore
@@ -4,3 +4,4 @@
 *.lock
 *.log
 *.mp3
+.wrangler

--- a/index.js
+++ b/index.js
@@ -1,4 +1,4 @@
-import { Router } from 'itty-router';
+import { Router, createCors, error, json } from 'itty-router';
 
 // import the routes
 import { chatHandler } from './routes/chat';
@@ -7,8 +7,15 @@ import { embeddingsHandler } from './routes/embeddings';
 import { transcriptionHandler, translationHandler } from './routes/audio';
 import { getImageHandler, imageGenerationHandler } from './routes/image';
 
+const { preflight, corsify } = createCors();
+
 // Create a new router
 const router = Router();
+
+// CORS, see https://itty.dev/itty-router/cors
+router
+	// embed preflight upstream to handle all OPTIONS requests
+	.all('*', preflight);
 
 // chat completion
 router.post('/chat/completions', chatHandler);
@@ -33,5 +40,17 @@ router.get('/images/get/:name', getImageHandler);
 router.all('*', () => new Response('404, not found!', { status: 404 }));
 
 export default {
-	fetch: router.handle,
+	fetch: request =>
+		router
+			.handle(request)
+
+			// transform unformed responses
+			.then(json)
+
+			// catch any errors
+			.catch(error)
+
+			// add CORS headers to all requests,
+			// including errors
+			.then(corsify),
 };

--- a/index.js
+++ b/index.js
@@ -40,12 +40,15 @@ router.get('/images/get/:name', getImageHandler);
 router.all('*', () => new Response('404, not found!', { status: 404 }));
 
 export default {
-	fetch: request =>
+	fetch: (request, env, ctx) =>
 		router
-			.handle(request)
+			.handle(request, env, ctx)
 
 			// catch any errors
-			.catch(error)
+			.catch(e => {
+				console.error(e);
+				return error(e);
+			})
 
 			// add CORS headers to all requests,
 			// including errors

--- a/index.js
+++ b/index.js
@@ -44,9 +44,6 @@ export default {
 		router
 			.handle(request)
 
-			// transform unformed responses
-			.then(json)
-
 			// catch any errors
 			.catch(error)
 

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
 	},
 	"dependencies": {
 		"@cloudflare/ai": "^1.0.15",
-		"itty-router": "^2.6.1"
+		"itty-router": "^4.2.2"
 	},
 	"devDependencies": {
 		"vitest": "^0.31.0",

